### PR TITLE
Ruby: Add additional sql sinks for ActiveRecord connection methods

### PR DESCRIPTION
**Pull request by joefarebrother**
Part of https://github.com/github/codeql-team/issues/2616 and https://github.com/github/codeql-team/issues/2559

### CODE REVIEW

File: ruby/ql/lib/change-notes/2024-02-15-activerecord_connection_sql_sinks [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
```
Comment (sidshank) [inline ruby/ql/lib/change-notes/2024-02-15-activerecord_connection_sql_sinks:1]: This file seems to have been mistakenly committed in addition to the change-note markdown file.

File: ruby/ql/lib/codeql/ruby/frameworks/ActiveRecord.qll [hunk 2]
```diff
@@ -200,7 +200,13 @@ private predicate sqlFragmentArgumentInner(DataFlow::CallNode call, DataFlow::No
   call = activeRecordQueryBuilderCall("annotate") and
   sink = call.getArgument(_)
   or
-  call = activeRecordConnectionInstance().getAMethodCall("execute") and
+  call =
+    activeRecordConnectionInstance()
+        .getAMethodCall([
```
Comment (alexrford) [inline ruby/ql/lib/codeql/ruby/frameworks/ActiveRecord.qll:205]: We can add [`create`](https://api.rubyonrails.org/classes/ActiveRecord/ConnectionAdapters/DatabaseStatements.html#method-i-create) (an alias for `insert`) to this.

I think we can also remove `select_update`? I can't find this as a query execution method.

**Review by alexrford — CHANGES_REQUESTED** [review]
Sorry for the delay in reviewing this! One minor comment, otherwise looks great.

**Review by hmac — APPROVED** [review]

**Comment by hmac** [comment]
Ah it looks like you need an explicit re-approval from @alexrford to merge this - I thought my approval would be enough. Sorry!

**Review by alexrford — APPROVED** [review]

**Pull request merged**